sdk/core/azure-core-amqp/src/main/java/com/azure/core/amqp/implementation/ReactorSession.java

```diff
@@ -428,7 +432,7 @@ protected ReactorReceiver createConsumer(String entityPath, Receiver receiver,
      *
      * @return A new instance of an {@link AmqpLink} with the correct properties set.
      */
-    protected Mono<AmqpLink> createProducer(String linkName, String entityPath, Duration timeout,
+    protected Mono<AmqpSendLink> createProducer(String linkName, String entityPath, Duration timeout,
```

Comment:
Change back to AmqpLink? and user cast it to concrete class if needed?

Reply:
yes, and there is one more reason, I keep the same pattern with the `createConsumer` method which also return an AmqpReceiveLink.